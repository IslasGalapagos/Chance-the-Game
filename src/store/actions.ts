/*
  This Source Code Form is subject to the terms of the Mozilla Public
  License, v. 2.0. If a copy of the MPL was not distributed with this
  file, You can obtain one at http://mozilla.org/MPL/2.0/.

  Copyright Â© 2018 Evgeny Sysoletin. All rights reserved.
*/

import {createStandardAction} from 'typesafe-actions';

export const setName = createStandardAction('SET_NAME')<string>();
export const setScore = createStandardAction('SET_SCORE')<[number, number]>();
export const setCoefficients = createStandardAction('SET_COEFFICIENTS')<
  [number, number]
>();
export const setTotalScore = createStandardAction('SET_TOTAL_SCORE')<
  [number, number]
>();
