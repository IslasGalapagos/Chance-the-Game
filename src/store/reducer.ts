/*
  This Source Code Form is subject to the terms of the Mozilla Public
  License, v. 2.0. If a copy of the MPL was not distributed with this
  file, You can obtain one at http://mozilla.org/MPL/2.0/.

  Copyright Â© 2018 Evgeny Sysoletin. All rights reserved.
*/

import {ActionType, getType} from 'typesafe-actions';
import * as actions from './actions';

type TupleNum = [number, number];

export interface State {
  readonly name: string;
  readonly score: TupleNum;
  readonly coefficient: TupleNum;
  readonly totalScore: TupleNum;
}

const initialState: Readonly<State> = {
  name: '',
  score: [0, 0],
  coefficient: [1, 1],
  totalScore: [0, 0]
};

type Actions = ActionType<typeof actions>;

const reducer = (state: State = initialState, action: Actions): State => {
  switch (action.type) {
    case getType(actions.setName):
      return {...state, ...{name: action.payload}};

    case getType(actions.setScore):
      return {...state, ...{score: action.payload}};

    case getType(actions.setCoefficient):
      return {...state, ...{coefficient: action.payload}};

    case getType(actions.setTotalScore):
      return {...state, ...{totalScore: action.payload}};

    default:
      return state;
  }
};

export default reducer;
