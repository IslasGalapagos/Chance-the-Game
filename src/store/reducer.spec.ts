/*
  This Source Code Form is subject to the terms of the Mozilla Public
  License, v. 2.0. If a copy of the MPL was not distributed with this
  file, You can obtain one at http://mozilla.org/MPL/2.0/.

  Copyright Â© 2018 Evgeny Sysoletin. All rights reserved.
*/

import reducer from './reducer';
import {setName, setScore, setCoefficients, setTotalScore} from './actions';

const initialState = reducer(undefined, {} as any);

describe('reducer', () => {
  it('has initial state', () => {
    expect(initialState).toMatchSnapshot();
  });

  it('sets name', () => {
    const action = setName('Diophantus');
    const state = reducer(initialState, action);
    expect(state.name).toEqual('Diophantus');
  });

  it('sets score', () => {
    const action = setScore([2, 1]);
    const state = reducer(initialState, action);
    expect(state.score).toEqual([2, 1]);
  });

  it('sets coefficients', () => {
    const action = setCoefficients([2, 1]);
    const state = reducer(initialState, action);
    expect(state.coefficients).toEqual([2, 1]);
  });

  it('sets totalScore', () => {
    const action = setTotalScore([50, 0]);
    const state = reducer(initialState, action);
    expect(state.totalScore).toEqual([50, 0]);
  });
});
