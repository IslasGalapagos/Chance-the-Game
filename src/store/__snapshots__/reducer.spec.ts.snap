// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer has initial state 1`] = `
Object {
  "coefficients": Array [
    0,
    0,
  ],
  "name": "",
  "score": Array [
    0,
    0,
  ],
  "totalScore": Array [
    0,
    0,
  ],
}
`;
